src/main/java/io/debezium/connector/yugabytedb/YugabyteDBConnectorConfig.java

```diff
@@ -1433,6 +1545,365 @@ private static int validateLogicalDecodingMessageExcludeList(Configuration confi
+                            retryCount, config.getInteger(MAX_CONNECTOR_RETRIES));
+                e.printStackTrace();
+
+                pauseBetweenRetries(config.getLong(CONNECTOR_RETRY_DELAY_MS));
+            }
+        }
+
+        if (exception != null) {
+            throw new DebeziumException(exception);
+        }
+
+        return null;
+    }
+
+    public static JdbcConfiguration getJdbcConfig(Configuration config) {
+            return JdbcConfiguration.copy(Configuration.empty())
+                    .with(JdbcConfiguration.DATABASE, config.getString(DATABASE_NAME))
+                    .with(JdbcConfiguration.HOSTNAME, config.getString(HOSTNAME))
+                    .with(JdbcConfiguration.PORT, config.getString(PORT))
+                    .with(JdbcConfiguration.USER, config.getString(USER))
+                    .with(JdbcConfiguration.PASSWORD, config.getString(PASSWORD))
+                    .withDefault(YugabyteDBConnectorConfig.MAX_RETRIES, config.getInteger(MAX_RETRIES))
+                    .with(YugabyteDBConnectorConfig.RETRY_DELAY_MS, config.getLong(RETRY_DELAY_MS))
+                    .build();
+    }
+
+    public static String fetchTableIncludeList(YugabyteDBConnection ybConnection, String publicationName) throws SQLException {
+        try (Connection connection = ybConnection.connection();
+              final Statement statement = connection.createStatement()) {
+            String query = "SELECT * FROM pg_publication_tables WHERE pubname = '" + publicationName + "';";
+            final ResultSet rs = statement.executeQuery(query);
+            String tableList = "";
+            while (rs.next()) {
+                String tableName = rs.getString("tablename");
+                String schemaName = rs.getString("schemaname");
+                tableList += schemaName + "." + tableName + ",";
+            }
+            return tableList.substring(0, tableList.length() - 1);
+        }
+    }
+
+    public static boolean shouldUsePublication(Configuration config) {
+        String streamId = config.getString(YugabyteDBConnectorConfig.STREAM_ID);
+
+        // For CQL tables as well as for the config not using Publication/Replication streamId will be non null
+        if (streamId == null || streamId.isEmpty()) {
+            return true;
+        }
+
+        try (YBClient ybClient = YBClientUtils.getYbClient(config)) {
+            GetDBStreamInfoResponse getDBStreamInfoResponse = ybClient.getDBStreamInfo(streamId);
+            // Note that we need to pass one table ID to the ListCDCStreamsRequest RPC call.
+            Objects.requireNonNull(getDBStreamInfoResponse.getTableInfoList().get(0));
+            ListCDCStreamsResponse resp;
+            resp = ybClient.listCDCStreams(
+                            getDBStreamInfoResponse.getTableInfoList().get(0).getTableId().toStringUtf8(),
+                            getDBStreamInfoResponse.getNamespaceId(), null);
```

Comment:
Why are we passing the table id in listCDCStreams? It will work but can be very confusing for readers. Ideally, we should just list out all CDC streams for the namespace.